Report address of the object to be freed to Tracy

mkykadir opened the pull request:
Garbage sweeping works per-page basis but allocation is object based. Thus, object addresses is reported to Tracy and freeing also needs to report object address. Previously it was reporting page addresses to Tracy during sweeping. This is the reason why it was passing with `COLLECT_PRINTF` debug flag.

Fixes #2778

Geod24 commented on source/agora/utils/gc/GC.d:
```diff
@@ -2483,12 +2484,13 @@ struct Gcx
                                     assert(core.bitop.bt(toFree.ptr, i));
 
                                     debug(COLLECT_PRINTF) printf("\tcollecting %p\n", p);
-                                    leakDetector.log_free(sentinel_add(p));
```
That means that `leakDetector` doesn't quite work right ?

mkykadir replied:
Leak detector is not checking for double allocations, I’ve found the problem by updating `log_malloc`. I ~can~ also send that with the PR.

Commit:
GC: Report allocationg without free'ing in LeakDetector

Geod24 commented:
Wouldn't it be better to just set `hasFinalizer` to `true`, always ? The idea is to minimize the diff with upstream, which is [here](https://github.com/dlang/druntime/tree/master/src/core/internal/gc).

Commit:
GC: Report address of the object to be freed to Tracy

Garbage sweeping works per-page basis but allocation is object based.
Thus, object addresses is reported to Tracy and freeing also needs to
report object address.

Geod24 commented:
Interestingly I just found this: https://github.com/dlang/druntime/commit/6558d800f8b4db57bb8954511b7c5987590a0480
